libbeat/api/routes.go

```diff
@@ -73,3 +97,57 @@ func prettyPrint(w http.ResponseWriter, data common.MapStr, u *url.URL) {
 		fmt.Fprintf(w, data.String())
 	}
 }
+
+type Routes struct {
+	routes map[string]handlerFunc
+	log    *logp.Logger
+	smux   *http.ServeMux
+	mux    sync.RWMutex
+}
+
+func (d *Routes) handle(w http.ResponseWriter, r *http.Request) {
+	d.mux.RLock()
+	defer d.mux.RUnlock()
```

Comment:
these locks are "heavy". Although we have an http server, this look will allow to have only one active connection at a time.